extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/util/MultiStageQueryContext.java

```diff
@@ -88,14 +94,30 @@ public static String getMSQMode(final QueryContext queryContext)
     );
   }
 
-  public static boolean isDurableStorageEnabled(final QueryContext queryContext)
+  public static boolean isDurableShuffleStorageEnabled(final QueryContext queryContext)
   {
     return queryContext.getBoolean(
         CTX_ENABLE_DURABLE_SHUFFLE_STORAGE,
         DEFAULT_ENABLE_DURABLE_SHUFFLE_STORAGE
     );
   }
 
+  public static boolean isComposedIntermediateSuperSorterStorageEnabled(final QueryContext queryContext)
+  {
+    return queryContext.getBoolean(
+        CTX_COMPOSED_INTERMEDIATE_SUPER_SORTER_STORAGE,
+        DEFAULT_COMPOSED_INTERMEDIATE_SUPER_SORTER_STORAGE
+    );
+  }
+
+  public static long getIntermediateSuperSorterStorageMaxLocalBytes(final QueryContext queryContext)
```

Comment:
If isComposedIntermediateSuperSorterStorageEnabled is enabled and getIntermediateSuperSorterStorageMaxLocalBytes is not set ie default value, what will happen?

Reply:
The default values for `getIntermediateSuperSorterStorageMaxLocalBytes` is Long.MAX_VALUE so effectively it will only use local disk as before.